staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -71,9 +71,9 @@ func WithAuthentication(handler http.Handler, auth authenticator.Request, failed
 		// authorization header is not required anymore in case of a successful authentication.
 		req.Header.Del("Authorization")
 
```

Comment:
TODO to check if the response has a valid audience if apiAuds is non-empty

Reply:
Added TODO here and in TokenReview storage.